shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1596,7 +1596,55 @@ - (void)startKeyBoardAnimation:(NSTimeInterval)duration {
 
   // Invalidate old vsync client if old animation is not completed.
   [self invalidateKeyboardAnimationVSyncClient];
-  [self setupKeyboardAnimationVsyncClient];
+
+  fml::WeakPtr<FlutterViewController> weakSelf = [self getWeakPtr];
```

Comment:
nits:
Instead of fml::WeakPtr, we can use __block typeof(self) blockSelf = self; Then avoid using `fml::scoped_nsobject` in the block

Reply:
Offline discussion, we will keep using WeakPtr because __block causes retain cycle in ARC and can be overlooked when the code base is moved to ARC.